stripe/src/main/java/com/stripe/android/paymentsheet/CurrencyFormatter.kt

```diff
@@ -7,47 +7,77 @@ import java.util.Locale
+         * - how do they separate thousand digits (i.e. France uses spaces)
+         * When you get the {@link NumberFormat#getCurrencyInstance()} you are getting all this information.
+         *
+         * Some fields not used here, but might be relevant in other scenarios:
+         * - positive and negative numbers
+         *
+         * Among other things the currency format of the amount might have decimal places
+         * even if the targetLocale's currency does not, so below we will do a sort of merge of
+         * the targetLocale and amount currency. We will start with the currency format of the
+         * targetLocale, this sets, most notably, the number decimal and thousands separators,
+         * where the currency symbol should be placed relative to the number. Then we find the
+         * currency symbol of the amount for the targetLocale, and use that. Finally, we set the
+         * number of decimal places used by the amount currency.
+         *
+         * See the [NumberFormat] for why we use a try block.
+         */
+        val currencyFormat = NumberFormat.getCurrencyInstance(targetLocale)
 
-        // We need to cast inside the try catch because most currencies are decimal formats but
-        // not all. See the official Google Docs for NumberFormat for more context.
         runCatching {
             val decimalFormatSymbols =
                 (currencyFormat as DecimalFormat).decimalFormatSymbols
-            decimalFormatSymbols.currency = currency
-            decimalFormatSymbols.currencySymbol = currencySymbol
+            decimalFormatSymbols.currency = amountCurrency
+            decimalFormatSymbols.currencySymbol = amountCurrency.getSymbol(targetLocale)
+            currencyFormat.minimumFractionDigits = amountCurrencyDecimalDigits
             currencyFormat.decimalFormatSymbols = decimalFormatSymbols
         }
 
         return currencyFormat.format(majorUnitAmount)
     }
 
-    private fun getDefaultFractionDigits(currency: Currency): Int {
+    private fun getDefaultDecimalDigits(currency: Currency): Int {
         /**
          * Handle special cases where the client's default fractional digits for a given currency
          * don't match the Stripe backend's assumption.
          */
-        return when (currency.currencyCode.toUpperCase(Locale.ROOT)) {
-            "AFN" -> 2 // Afghanistan afghani
-            "ALL" -> 2 // Albanian lek
-            "AMD" -> 2 // Armenian dram
-            "COP" -> 2 // Colombia peso
-            "IDR" -> 2 // Indonesian rupiah
-            "ISK" -> 2 // Icelandic króna
-            "PKR" -> 2 // Pakistani rupee
-            "LBP" -> 2 // Lebanese pound (a.k.a. lira)
-            else -> currency.defaultFractionDigits
+        return if (SERVER_DECIMAL_DIGITS_2.contains(currency.currencyCode.toUpperCase(Locale.ROOT))) {
+            2
+        } else {
+            currency.defaultFractionDigits
         }
```

Comment:
This is totally fine, but you could optionally do this to be able to easily support other decimal values in the future

```
        return SERVER_DECIMAL_DIGITS
            .filter { entry ->
                entry.key.contains(currency.currencyCode.toUpperCase(Locale.ROOT))
            }.map {
                it.value
            }.firstOrNull() ?: currency.defaultFractionDigits
```

and then below
```
        internal val SERVER_DECIMAL_DIGITS =
            mapOf(
                setOf("UGX", "AFN", "ALL", "AMD", "COP", "IDR", "ISK", "PKR", "LBP") to 2
            )
```